consensus/consensus-types/src/block.rs

```diff
@@ -309,3 +311,17 @@ impl<'de, T: DeserializeOwned + Serialize> Deserialize<'de> for Block<T> {
         })
     }
 }
+
+impl<T> From<&Block<T>> for BlockMetadata {
```

Comment:
should we merge block info and block metadata? now that we're passing the blockinfo back via the ledgerinfo it seems only reasonable to merge these two data structures as well.

Reply:
they carry different different fields right? but those naming are really confusing , how about OnChainBlock?

Reply:
I don't think this is part of this PR, but maybe it is worth contemplating if we should merge them?

Reply:
Naming is interesting now that we have BlockInfo in libra-types

Reply:
sure, we should definitely discuss now we have BlockData, BlockInfo, BlockMetadata lol

Reply:
although BlockInfo is really a ExecutedBlockInfo, it can't be generated before the execution while BlockMetadata needs to be generated before